internal/server/evaluation/evaluation.go

```diff
@@ -0,0 +1,84 @@
+package evaluation
+
+import (
+	"context"
+	"errors"
+
+	errs "go.flipt.io/flipt/errors"
+	fliptotel "go.flipt.io/flipt/internal/server/otel"
+	"go.flipt.io/flipt/internal/storage"
+	"go.flipt.io/flipt/rpc/flipt"
+	rpcEvaluation "go.flipt.io/flipt/rpc/flipt/evaluation"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+// Variant evaluates a request for a multi-variate flag and entity.
+func (s *Server) Variant(ctx context.Context, v *rpcEvaluation.EvaluationRequest) (*rpcEvaluation.VariantEvaluationResponse, error) {
+	var ver = &rpcEvaluation.VariantEvaluationResponse{}
+
+	flag, err := s.store.GetFlag(ctx, v.NamespaceKey, v.FlagKey)
+	if err != nil {
+		var errnf errs.ErrNotFound
+
+		if errors.As(err, &errnf) {
+			ver.Reason = rpcEvaluation.EvaluationReason_FLAG_NOT_FOUND_EVALUATION_REASON
+			return ver, err
+		}
+
+		ver.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+		return ver, err
+	}
+
+	if flag.Type != flipt.FlagType_VARIANT_FLAG_TYPE {
+		ver.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+		return ver, errs.ErrInvalidf("flag type %s invalid", flipt.FlagType_name[int32(flag.Type)])
+	}
```

Comment:
I think we're in a situation now where we're vaidating the flag exists twice. Once in and once out of the evaluator.

Ultimately, having the service layer do it is good I think. At-least in terms of fetching the flag.
Instead, I think we should make `Evaluate` take an entire `*flipt.Flag` instead of just the key.

Then I think evaluate should handle validating the flag type is appropriate. So that should move into that layer.

For example, the evaluator interface method would become:
```go
// MultiVariateEvaluator is an abstraction for evaluating a flag against a set of rules for multi-variate flags.
type MultiVariateEvaluator interface {
	Evaluate(context.Context, *flipt.Flag, *flipt.EvaluationRequest) (*flipt.EvaluationResponse, error)
}
```

The the implementation would do a fail early if the type doesnt match.

Reply:
Also, is it possible to move that interface into this package? or into the parent server package, wherever doesn't create the cycle.